include/seqan3/search/search_result.hpp

```diff
@@ -0,0 +1,200 @@
+        }
+        return cursor_;
+    }
+
+    /*!\brief Returns the reference id where the query was found.
+     *
+     * \details
+     * The reference id is an arithmetic value that corresponds to the index of the reference text in the index.
+     * The order is determined on construction of the index.
+     */
+    constexpr reference_id_type reference_id() const noexcept(!(std::same_as<reference_id_type, detail::empty_type>))
+    {
+        if constexpr (std::same_as<reference_id_type, detail::empty_type>)
+        {
+            throw std::logic_error{"You tried to access the reference id but it was not selected in the output "
+                                   "configuration of the search."};
+        }
+        return reference_id_;
+    }
+
+    //!\brief Returns the reference begin positions where the query was found in the reference text (at `reference id`).
+    constexpr reference_begin_pos_type reference_begin_pos() const
+        noexcept(!(std::same_as<reference_begin_pos_type, detail::empty_type>))
+    {
+        if constexpr (std::same_as<reference_begin_pos_type, detail::empty_type>)
+        {
+            throw std::logic_error{"You tried to access the reference begin position but it was not selected in the "
+                                   "output configuration of the search."};
+        }
+        return reference_begin_pos_;
+    }
+    //!\}
+
+    /*!\name Comparison
+     * \{
+     */
+    //!\brief Returns whether `lhs` and `rhs` are the same.
+    friend bool operator==(search_result const & lhs, search_result const & rhs) noexcept
+    {
+        bool equality = lhs.query_id_ == rhs.query_id_;
+        if constexpr (!std::is_same_v<cursor_type, detail::empty_type>)
+            equality &= lhs.cursor_ == rhs.cursor_;
+        if constexpr (!std::is_same_v<reference_id_type, detail::empty_type>)
+            equality &= lhs.reference_id_ == rhs.reference_id_;
+        if constexpr (!std::is_same_v<reference_begin_pos_type, detail::empty_type>)
+            equality &= lhs.reference_begin_pos_ == rhs.reference_begin_pos_;
+
+        return equality;
+    }
+
+    //!\brief Returns whether `lhs` and `rhs` are not the same.
+    friend bool operator!=(search_result const & lhs, search_result const & rhs) noexcept
+    {
+        return !(lhs == rhs);
+    }
+    //!\}
+};
+
+/*!\brief Print the seqan3::search_result to seqan3::debug_stream.
+ * \tparam char_t The underlying character type of the seqan3::debug_stream_type.
+ * \tparam search_result_t A specialization of seqan3::search_result.
+ * \param ds The stream.
+ * \param res The search result to print.
+ * \relates seqan3::debug_stream_type
+ */
+template <typename char_t, typename search_result_t>
+//!\cond
+    requires detail::is_type_specialisation_of_v<remove_cvref_t<search_result_t>, search_result>
+//!\endcond
+inline debug_stream_type<char_t> & operator<<(debug_stream_type<char_t> & ds, search_result_t && res)
```

Comment:
Well, I am not really convinced about this. But it is also only debug stuff. 
In general, please use proper names for the variables instead of the abbreviations.